app/controllers/application_controller.rb

```diff
@@ -387,6 +388,16 @@ def report_data
     }
   end
 
+  def options_from_session(options)
```

Comment:
Sorry, my point about `where_clause` was not that you don't need it, just that you don't need to pass it around via params since it's always the same clause, you just need the info on whether to use it or not.